### DISCUSSION TITLE
How to get the Monday time of current week?

### DISCUSSION THREAD

**Discussion by hustcer** [nushell/nushell#4195] [Q&A] [answered]
I want to get the Monday time of current week like this:

```nu
def 'get-monday' [] {
  let weekDay = ([(date now)] | dataframe to-df | dataframe get-weekday).0
  let today = (date to-table|select year month day)
  echo ($'($today.year)-($today.month)-($today.day - $weekDay)' | str to-datetime | date format '%Y-%m-%d %H:%M:%S')
}
```
For now, the result would be `2021-12-06 00:00:00`.
It works sometime, but not always, as `($today.day - $weekDay)` could be a negative number, so how to do this in the right way? Thanks.

**Comment by hustcer** [comment]
And any way to convert a string to duration? Thanks

**Reply by kubouch** [reply]
You can use `str to-datetime` to convert string to a date, then you can subtract the dates to get a duration. Not sure if it's possible to convert string to duration directly. Seems like not, but I haven't worked with these date commands much.

For the `($today.day - $weekDay)`, you can perhaps wrap it in `if` and check for negative result?

**Reply by hustcer** [reply]
I haven't found a way to convert string to duration, maybe we could add this feature in nushell?
Use `if` and check for negative result should work, but it's not elegant enough, that will be my last choice.

**Reply by kubouch** [reply]
I guess it could be added. If you want to work on it, it should be added to the engine-q project since that is going to be the Nushell: https://github.com/nushell/engine-q

**Reply by kubouch** [reply]
You could create a `satsub` custom command that would abstract away the `if` block for example...

**Reply by hustcer** [reply]
Thanks, I will try it later

**Comment by hustcer** [comment, marked as answer]
I think this should be a better solution to get the beginning time of Monday:
```nu
# Get the begin time of monday, like 2021-12-06 00:00:00
def 'get-monday' [] {
  let today = (date to-table|select year month day)
  # Currently convert string to duration is not supported
  let durations = [0day, 1day, 2day, 3day, 4day, 5day, 6day]
  let weekDay = ([(date now)] | dataframe to-df | dataframe get-weekday).0
  let beginOfToday = ($'($today.year)-($today.month)-($today.day)' | str to-datetime)
  echo (($beginOfToday - ($durations | nth $weekDay)) | date format '%Y-%m-%d %H:%M:%S')
}
```

**Comment by solodov** [comment]
I ran into similar problem, but looking for next Monday, this brought me here. The solution I found is this:

```nu
def "get-monday" [] {
  (seq date -r --days 7 |
   into datetime | 
   where {|e| ($e | date format "%u") == "1"}).0 | date format "%Y-%m-%d %H:%M:%S"
}
```

Question has been answered, so adding this here for posterity, maybe someone else will find this useful.

**Reply by theAkito** [reply]
For fun, I created a remix of your version.

```nu
# Get Monday of current week.
export def "get-current-monday" [] {
  seq date -r --days 7
  | into datetime
  | where { |e| ($e | date format "%u") == "1"}
  | take
  | date format "%Y-%m-%dT%H:%M:%SZ"
}
```
